tests/agent_server/test_server_agent.py

```diff
@@ -3092,7 +3093,8 @@ async def test_deploy_no_code(resource_container, client, clienthelper, environm
     await clienthelper.put_version_simple(resources, version)
 
     await _wait_until_deployment_finishes(client, environment, version)
-    # The resource state and its logs are not set atomically. This call prevents a race condition.
+    # The version, resource state and its logs are not set atomically. This call prevents a race condition.
+    await wait_until_resource_done_state(client, environment, resource_id)
```

Comment:
This cannot be the cause of the race condition. `_wait_until_deployment_finishes()` only returns when all the resources are in a non-transient state, while the test case fails because a resource is in the "deploying" state, which is a transient state.

The fact that the resource goes to deploying again, indicates that the deploy or the repair timer has fired. The `autostarted_agent` fixture, [disables the deploy and repair timer](https://github.com/inmanta/inmanta-core/blob/09930d94f66ab663282158b8ea1d74450af96c5d/tests/conftest.py#L659). To activate this setting, the agent needs to be restarted. It seems that [the logic to restart the agent is only present in the code path of the API](https://github.com/inmanta/inmanta-core/blob/09930d94f66ab663282158b8ea1d74450af96c5d/src/inmanta/server/services/environmentservice.py#L203), but this test case doesn't go over the API. It goes directly to the database layer.